### ISSUE TITLE
TracingService breaks Ambassador 1.8.0

### ISSUE THREAD

**Issue by dliraf** [datawire/ambassador#3019]
**Describe the bug**
"100%" cpu usage and no availability when using TracingService with Ambassador 1.8.0 (Helm chart 6.5.9)  

**To Reproduce**
```
apiVersion: helm.fluxcd.io/v1
kind: HelmRelease
metadata:
  name: ambassador
  namespace: ambassador
spec:
  releaseName: ambassador
  chart:
    repository: https://www.getambassador.io
    name: ambassador
    version: 6.5.9
  values:
    image:
      repository: quay.io/datawire/ambassador
    enableAES: false
```
```
apiVersion: getambassador.io/v2
kind:  TracingService
metadata:
  namespace: ambassador
  name:  tracing
spec:
  service: "opentelemetry-collector.opentelemetry:9411"
  driver: zipkin
  config: {}
```

**Versions (please complete the following information):**
 - Ambassador: 1.8.0
 - Kubernetes environment: Google GKE 1.16.13-gke.401

**Additional context**
Works when using Helm chart 6.5.3 (Ambassador 1.7.0)  
I.e. low cpu and the traces are being sent

**Comment by esmet** [comment]
Hey @dliraf - Thanks for the report.

I was able to reproduce this issue on 1.8.0 using the TracingService you provided and I'm working on the fix. The underlying issue appears to be with service names whose internal representation exceeds 40 characters and gets compressed before it is referenced in Envoy configuration.

The best workaround right now would be to use a `service:` string that is shorter. Could you try creating another, shorter service name for `opentelemetry-collector`? My recommendation would be to name the new shortened service `otc` and keep it in the `opentelemetry` namespace. The resulting service string would then be `otc.opentelemetry`. I verified in local testing that this service name is short enough to elide the requisite compression step that exposes the underlying bug.

**Comment by khussey** [comment]
This is fixed in the 1.8.1 release, which is now available.
